### DISCUSSION TITLE
KafkaConnect JSON logs

### DISCUSSION THREAD

**Discussion by mathieux51** [strimzi/strimzi-kafka-operator#5049] [Q&A] [answered]
Howdy, 

I'm trying to setup JSON logging for `kind: KafkaConnect`, this is how it looks:
```
 logging:
     type: inline
     loggers:
       log4j.appender.CONSOLE: org.apache.log4j.ConsoleAppender
       log4j.appender.CONSOLE.layout: net.logstash.log4j.JSONEventLayoutV1
       log4j.rootLogger: INFO, CONSOLE
```
Now I'm getting this error:
```
log4j:ERROR Could not instantiate class [net.logstash.log4j.JSONEventLayoutV1].
java.lang.ClassNotFoundException: net.logstash.log4j.JSONEventLayoutV1
    at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:264)
    at org.apache.log4j.helpers.Loader.loadClass(Loader.java:198)
    at org.apache.log4j.helpers.OptionConverter.instantiateByClassName(OptionConverter.java:327)
    at org.apache.log4j.helpers.OptionConverter.instantiateByKey(OptionConverter.java:124)
    at org.apache.log4j.PropertyConfigurator.parseAppender(PropertyConfigurator.java:797)
    at org.apache.log4j.PropertyConfigurator.parseCategory(PropertyConfigurator.java:768)
    at org.apache.log4j.PropertyConfigurator.configureRootCategory(PropertyConfigurator.java:648)
    at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:514)
    at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:580)
    at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:526)
    at org.apache.log4j.LogManager.<clinit>(LogManager.java:127)
    at org.slf4j.impl.Log4jLoggerFactory.<init>(Log4jLoggerFactory.java:66)
    at org.slf4j.impl.StaticLoggerBinder.<init>(StaticLoggerBinder.java:72)
    at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
    at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
    at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
    at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:412)
    at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:357)
    at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:383)
    at org.apache.kafka.connect.cli.ConnectDistributed.<clinit>(ConnectDistributed.java:54)
```

I'm using this docker image `strimzi/kafka-connect:0.11.4-kafka-2.1.0`. 
Any idea how I can fix this `ClassNotFoundException`?
Thanks for in advance 🙏

**Comment by scholzj** [comment]
Are you sure JSON logging was supported in 0.11? If the class is missing, it looks like it might not be. You would need to build your custom images to add it.

**Reply by mathieux51** [reply]
Building a custom image sounds good, even using a greater version. Any idea where I can find those missing files and where I should put them? (Not a Java expert 😅)

**Reply by scholzj** [reply]
You can download it from Maven repositories (https://mvnrepository.com/artifact/net.logstash.log4j/jsonevent-layout/1.7), but you would also need to download all the different dependencies. In later Strimzi versions we have an automation for that, but if you want to stick with 0.11, you would need to go through them manually and add them to your image.

**Reply by mathieux51** [reply]
Looks like I'm missing something here! When I check https://hub.docker.com/r/strimzi/kafka-connect/tags?page=1&ordering=last_updated the version `strimzi/kafka-connect:0.11.4-kafka-2.1.0` is the most recent.

**Reply by scholzj** [reply]
The images names changed over time:
* We now use Quay.io: https://quay.io/organization/strimzi
* The `kafka-connect` image is not used for a long time ... the is just `kafka` now

**Reply by mathieux51** [reply]
@scholzj thanks a lot 🙏. I can now see JSON logs.

**Comment by mathieux51** [comment, marked as answer]
TLDR: I was using the wrong base image, this worked for me:
```
kind: KafkaConnect
spec:
  image: quay.io/strimzi/kafka:0.22.1-kafka-2.6.0
  logging:
    loggers:
      log4j.appender.CONSOLE: org.apache.log4j.ConsoleAppender
      log4j.appender.CONSOLE.layout: net.logstash.log4j.JSONEventLayoutV1
      log4j.rootLogger: INFO, CONSOLE
```
